src/Avalonia.Visuals/Media/FormattedText.cs

```diff
@@ -10,128 +10,109 @@ namespace Avalonia.Media
     /// <summary>
     /// Represents a piece of text with formatting.
     /// </summary>
-    public class FormattedText : AvaloniaDisposable
+    public class FormattedText
```

Comment:
The problem is that GC is unable to correctly determine the amount of memory occupied by texts, bitmaps, geometries, etc. So we might need some kind of reference counting wrapper anyway.